### PR TITLE
Implement support for CoAP over tcp

### PR DESCRIPTION
**Pull request by petardimitrijevic**
This pull request implements CoAP over TCP.

### CODE REVIEW

File: libcoap/src/transport/tcp.rs [hunk 1]
```diff
@@ -1,13 +1,86 @@
...
+
 #[cfg(feature = "tcp")]
-pub struct CoapTcpEndpoint {}
+#[derive(Debug)]
+pub struct CoapTcpEndpoint {
+    raw_endpoint: *mut coap_endpoint_t,
+}
+
+impl CoapTcpEndpoint {
+    /// Creates a new CoapUdpEndpoint and binds it to the supplied SocketAddr.
```
Comment (falko17) [inline libcoap/src/transport/tcp.rs:23]:
```suggestion
    /// Creates a new CoapTcpEndpoint and binds it to the supplied SocketAddr.
```

File: libcoap/tests/tcp_client_server_test.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * dtls_client_server_test.rs - Tests for UDP clients+servers.
```
Comment (falko17) [inline libcoap/tests/tcp_client_server_test.rs:3]:
```suggestion
 * dtls_client_server_test.rs - Tests for TCP clients+servers.
```

**Comment by pulsastrix** [comment]
Thanks for the PR!
LGTM aside from what @falko17 already mentioned.

Regarding the test failures: It seems that rust-lang/rust#109044 is the cause.
I have pushed a commit that switches the test jobs over to nightly, so rebasing the PR branch on main should fix the checks.

**Comment by petardimitrijevic** [comment]
I have fixed the comments. The changes are pushed.

> Thanks for the PR! LGTM aside from what @falko17 already mentioned.
> 
Glad that I could help.

**Review by pulsastrix — APPROVED** [review]

**Review by falko17 — APPROVED** [review]

**Comment by petardimitrijevic** [comment]
I've rebased the branch on main, so that it includes @pulsastrix switch to nightly toolchain.
The pipeline should pass with this commit.

**Comment by JKRhb** [comment]
@pulsastrix @falko17 I guess this PR should now be ready to be merged, right? Or does the coverage job need to be fixed first?

**Comment by pulsastrix** [comment]
@JKRhb the coverage job failure is due to the fact that this PR comes from a fork, which causes the CI pipeline to have insufficient permissions to create comments in the main repository.
Will merge now.

**Pull request merged**